### PR TITLE
{Core} Enrich user-agent for az rest

### PR DESCRIPTION
**Pull request by houk-ms**
**Description**  

This PR adds more properties into az rest user-agent, making it align with the python-sdk based requests. This is suggested by our PM Arun so that we can get the python version and os info from the user agents.

Currently, a common user agent from az rest based requests could be
```
AZURECLI/2.11.1 (RPM)
```
a common user agent from python-sdk based requests could be
```
python/3.8.2 (Windows-10-10.0.18362-SP0) msrest/0.6.18 msrest_azure/0.6.3 azure-mgmt-resource/10.2.0 Azure-SDK-For-Python AZURECLI/2.12.1 (PIP)
```

The PR adds the following properties into az rest user agents:
- python-version
- os detail

These properties won't be added because they are not the dependencies of az rest.
- msrest version
- msrest_azure version

**Testing Guide**  

Run the az rest command, e.g,
```
az rest --method GET --uri 'https://graph.microsoft.com/v1.0/applications/b4e4d2ab-e2cb-45d5-a31a-98eb3f364001' --debug
``` 
Previously, the userAgent is
```
AZURECLI/2.12.1 (PIP)
```
Now, the userAgent is
```
python/3.8.2 (Windows-10-10.0.18362-SP0) AZURECLI/2.12.1 (PIP)
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/util.py [hunk 1]
```diff
@@ -1077,6 +1077,22 @@ def get_az_user_agent():
     return ' '.join(agents)
 
 
+def get_az_rest_user_agent():
+    """Get User-Agent for az rest calls"""
+
+    from msrest import __version__ as msrest_version
+    from msrestazure import __version__ as msrestazure_version
+
+    agents = ['python/{}'.format(platform.python_version()),
```
Comment (yungezz) [inline src/azure-cli-core/azure/cli/core/util.py:1086]: is there any existing common util for this?

**Review by yungezz — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/util.py (hunk 1 above)
Comment (houk-ms) [reply src/azure-cli-core/azure/cli/core/util.py:1086]: I am searching for that, too. Will replace it if I could find the utils.

Comment (MyronFanQiu) [reply src/azure-cli-core/azure/cli/core/util.py:1086]: There should be a similar function in the python sdk, not in azure cli or azure cli core. Just curious, I thought we would remove the dependency of msrest and msrestazure in the future. Is it necessary to add msrest and msrestazure?

File: src/azure-cli-core/azure/cli/core/util.py [hunk 2]
```diff
@@ -1077,6 +1077,22 @@ def get_az_user_agent():
     return ' '.join(agents)
 
 
+def get_az_rest_user_agent():
+    """Get User-Agent for az rest calls"""
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/util.py:1081]: Consider add an example of the final result.

File: src/azure-cli-core/azure/cli/core/util.py [hunk 3]
```diff
@@ -1077,6 +1077,22 @@ def get_az_user_agent():
...
+
+    from msrest import __version__ as msrest_version
+    from msrestazure import __version__ as msrestazure_version
+
+    agents = ['python/{}'.format(platform.python_version()),
+              '({})'.format(platform.platform()),
+              'msrest/{}'.format(msrest_version),
+              'msrest_azure/{}'.format(msrestazure_version),
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/util.py:1089]: `az rest` never relies on `msrest` and `msrest_azure`. I doubt if we should add them.

Comment (houk-ms) [reply src/azure-cli-core/azure/cli/core/util.py]: I agree, removed.

File: src/azure-cli-core/azure/cli/core/util.py (hunk 1 above)
Comment (houk-ms) [reply src/azure-cli-core/azure/cli/core/util.py:1083]: I agree, removed.

File: src/azure-cli-core/azure/cli/core/util.py (hunk 2 above)
Comment (houk-ms) [reply src/azure-cli-core/azure/cli/core/util.py:1081]: sure

**Comment by houk-ms** [comment]
An exmaple of the az rest UserAgent now: `python/3.8.2 (Windows-10-10.0.18362-SP0) AZURECLI/2.12.1 (PIP)`

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**